src/mds/Beacon.cc

```diff
@@ -495,6 +495,24 @@ void Beacon::notify_health(MDSRank const *mds)
     MDSHealthMetric m(MDS_HEALTH_CACHE_OVERSIZED, HEALTH_WARN, css->strv());
     health.metrics.push_back(m);
   }
+
+  // Report laggy client(s) due to laggy OSDs
+  {
+    auto&& laggy_clients = mds->server->get_laggy_clients();
+    if (laggy_clients.size()) {
```

Comment:
I have to fetch laggy_clients from Server, this helper looks better rather than `mds->server->laggy_clients` and it is also used to clear set before moving to `find_idle_sessions` and `evict_cap_revoke_non_responders`

Reply:
That's fine. My point was to use `empty()` rather than `size()`.